.buildkite/pipeline.yml

```diff
@@ -11,43 +11,55 @@ env:
 
 steps:
   - label: hlint
+    key: hlint
     if: &not_scheduled_autodoc
       build.branch != "autodoc/master" && build.branch != "master" && build.source != "schedule"
     commands:
     - nix run -f ci.nix pkgs.hlint -c
         ./scripts/lint.sh
 
   - label: reuse lint
+    key: reuse_lint
     if: *not_scheduled_autodoc
     commands:
     - nix run -f ci.nix pkgs.reuse -c
         reuse lint
 
   - label: check trailing whitespace
+    key: check_trailing_whitespace
     if: *not_scheduled_autodoc
     commands:
     - .buildkite/check-trailing-whitespace.sh
 
   - label: crossref-verify
+    key: crossref-verify
     if: *not_scheduled_autodoc
     commands:
     - nix run -f ci.nix crossref-verifier -c
         crossref-verify --mode local-only --config ./.crossref-verifier.yaml
-    soft_fail: true  # TODO: remove
+    soft_fail: true
 
   - label: build
+    key: build
     if: *not_scheduled_autodoc
+    depends_on:
+      - hlint
+      - reuse_lint
+      - check_trailing_whitespace
+      - crossref-verify
```

Comment:
AFAIK `hlint` and `build` share some work, so it's good to let the latter depend on the former (to not repeat such work), however there shouldn't be problems with the rest.

So I think this should just depend on `hlint` and not on the others.